cdf_dump/src/main/scala/cognite/spark/cdfdump/Main.scala

```diff
@@ -0,0 +1,162 @@
+
+  val readOptions = props[String]('R', "Spark read options. You can use any option supported by CDF Spark Data Source, see: https://github.com/cognitedata/cdp-spark-datasource/#common-options")
+  val writeOptions = props[String]('W', "Spark write options. You can use any option supported in your selected output format. For example `-f csv -WincludeHeader=true` to write CSV with headers.")
+
+  val outputPartitions = opt[Int]("out-partitions", argName = "number", descr = "Number of output partitions for text formats. By default 1, increase for better performance, but fragmented output.", default = Some(1))
+  val maxRetries = opt[Int]("max-retries", argName = "number", descr = "Set to 0 for debugging, to improve responsibility")
+  group("Advanced Spark options").append(readOptions, writeOptions, outputPartitions, maxRetries)
+
+  verify()
+}
+
+object Main extends App {
+  val logger = getLogger
+  val a = new Conf(args)
+
+  val outDir = a.outDir()
+
+  if (a.clearOutDir()) {
+    if (Files.exists(Path.of(outDir))) {
+      if (Files.isDirectory(Path.of(outDir))) {
+        logger.info(s"Removing all items in $outDir")
+        Files.list(Path.of(outDir)).forEach(f => FileUtils.deleteQuietly(f.toFile))
+      } else {
+        logger.info(s"Removing file $outDir")
+        Files.delete(Path.of(outDir))
+      }
+    }
+  }
+
+  lazy val helper = new SparkHelper(
+    a.writeOptions,
+    a.readOptions,
+    outDir,
+    a.format(),
+    a.columns.toOption,
+    a.excludeColumn.getOrElse(List.empty),
+    a.filter.toOption,
+    a.outputPartitions.toOption,
+    a.maxRetries.toOption
+  )
+
+  a.raw().foreach(table => {
+    val Array(db, t) = table.split("[.]", 2)
+    log(s"Raw table $db.$t", helper.saveRaw(db, t))
+  })
+
+  if (a.assets() || a.allClean()) {
+    log("Assets", helper.saveClean("assets"))
+  }
+
+  if (a.events() || a.allClean()) {
+    log("Events", helper.saveClean("events"))
+  }
+
+  if (a.timeseries() || a.allClean()) {
+    log("Time series", helper.saveClean("timeseries"))
+  }
+
+  if (a.relationships() || a.allClean()) {
+    log("Relationships", helper.saveClean("relationships"))
+  }
+
+  if (a.files() || a.allClean()) {
+    log("Files", helper.saveClean("files"))
+  }
+
+  if (a.sequences() || a.allClean()) {
+    log("Sequences", helper.saveClean("sequences"))
+  }
+
+  if (a.labels() || a.allClean()) {
+    log("Labels", helper.saveClean("labels"))
+  }
+
+  if (a.datasets() || a.allClean()) {
+    log("Datasets", helper.saveClean("datasets"))
+  }
+
+  def log(c: String, action: => Unit): Unit = {
+    println(s"Downloading $c")
+    val t0 = System.nanoTime()
+    action
+    val t1 = System.nanoTime()
+    val time = (t1 - t0) / 1000 / 1000
+    //val time = java.time.Duration.ofNanos(t1 - t0)
+    // println(s"Loaded $c in ${DurationFormatUtils.formatDurationWords(time, true, true)}")
+    println(s"Loaded $c in ${DurationFormatUtils.formatDurationHMS(time)}")
```

Comment:
Prints can be removed?
and why do we use print here instead of logger?

Reply:
hmm, because I don't know how to configure the logger. It didn't work for some reason :|

Reply:
sorry i left the review unfinished, i'll complete it 😅